src/com/thunder_cut/netio/DataReceiver.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * ParticipantsImageReceiver.java
+ * Author : Hyeokwoo Kwon
+ * Created Date : 2020-02-01
+ */
+package com.thunder_cut.netio;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.SocketChannel;
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+/**
+ * Receives other participants' data
+ * <p>
+ */
+public class DataReceiver implements Runnable {
+
+    private enum HeaderItem {
+        TYPE,
+        SRC_ID,
+        DST_ID,
+        DATA_SIZE;
+    }
+
+    private SocketChannel socketChannel;
+    private DataUnwrapper receivedData;
+
+    private BiConsumer<Integer, byte[]> drawImage;
+
+    private Map<Character, BiConsumer<Integer, byte[]>> dataType;
+
+    @Override
+    public void run() {
+
+        dataType = new HashMap<Character, BiConsumer<Integer, byte[]>>();
+
+        dataType.put(DataType.IMG.type, drawImage);
+//        dataType.put(DataType.MSG, ***);
+//        dataType.put(DataType.CMD, ***);
+
+        while (true) {
+            receiveData();
+
+            dataType.get(receivedData.dataType).accept(receivedData.srcID, receivedData.data.array());
+        }
+    }
+
+    public void receiveData() {
+
+        try {
+            Map<HeaderItem, Integer> header = readHeader();
```

Comment:
You don't use Map instance. put readHeader() into readData()    
readData(readHeader));